docs/interop.md

```diff
@@ -0,0 +1,51 @@
+# Interoperability via Scripting
+
+Token SDK supports interoperability, such as HTLC mechanism (see related section), via scripting. 
+It allows spending a token to a script by encoding the script in the `Owner` field of a `Token`, and the different drivers are capable of interpreting the owner as a script.
+
+## HTLC (exchange)
+
+HTLC (Hash Time Locked Contract) is a token transfer that use hashlocks and timelocks to require that the recipient of a token either acknowledge receiving the token prior to a deadline by generating cryptographic proof or forfeit the ability to claim the token, returning it to the sender.
+
+### HTLC script
+
+The HTLC (exchange) script encodes the details of the exchange, the identities of the sender and the recipient of the token, a deadline, and hashing information.
+
+```go
+// Script contains the details of an exchange
+type Script struct {
+    Sender    view.Identity
+    Recipient view.Identity
+    Deadline  time.Time
+    HashInfo  HashInfo
+}
+
+// HashInfo contains the information regarding the hashing
+type HashInfo struct {
+    Hash         []byte
+    HashFunc     crypto.Hash
+    HashEncoding encoding.Encoding
+}
+```
+
+## Interoperability services
+
+Some interoperability services, located in `token/services/interop`, which are responsible for assembling the token transaction, managing the transaction lifecycle, and so on, are the same as the `Token Transaction Services`. 
+Other services are script specific. 
+
+For example, the token transaction assembling service enables appending exchange, claim, or reclaim actions to the token request of the transaction. All of these actions translate into a transfer action. 
+The interop `Wallet` service supports listing expired tokens, whose deadline have passed, and listing tokens with a desired matching preimage.
```

Comment:
How can I get this wallet?

Reply:
Should we also list the API? A link to the code may also work?

Reply:
added a location
is it enough?